tests/src/test/java/org/wso2/micro/gateway/tests/services/MutualSSLTestCase.java

```diff
@@ -0,0 +1,210 @@
+            logger.info("An UnknownHostException  occurred: " + e);
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            Logger logger = null;
+            logger.info("An IOException occurred: " + e);
+
+        }
+    }
+
+
+    @Test(description = "mutual SSL is filed due to bad certificate")
+    public void mutualSSLfail() throws Exception {
+
+        String trustStorePath = getClass().getClassLoader()
+                .getResource("keyStores" + File.separator + "ballerinaTruststore.p12").getPath();
+        Properties systemProps = System.getProperties();
+        systemProps.put("javax.net.debug", "ssl");
+        systemProps.put("javax.net.ssl.trustStore", trustStorePath);
+        systemProps.put("javax.net.ssl.trustStorePassword", "ballerina");
+        System.setProperties(systemProps);
+        SSLContext sslcontext;
+        KeyStore keyStore;
+        final char[] P12_PASSWORD = "ballerina".toCharArray();
+        final char[] KEY_PASSWORD = "ballerina".toCharArray();
+
+        try {
+            String keyPath = getClass().getClassLoader()
+                    .getResource("keyStores" + File.separator + "mtsltestFail.p12").getPath();
+            final InputStream is = new FileInputStream(keyPath);
+            keyStore = KeyStore.getInstance("pkcs12");
+            keyStore.load(is, P12_PASSWORD);
+            final KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            kmf.init(keyStore, KEY_PASSWORD);
+            sslcontext = SSLContext.getInstance("TLS");
+            sslcontext.init(kmf.getKeyManagers(), null, new java.security.SecureRandom());
+        } catch (Exception ex) {
+            throw new IllegalStateException("Failure initializing default SSL context", ex);
+        }
+
+        SSLSocketFactory sslsocketfactory = sslcontext.getSocketFactory();
+        try {
+
+            URL url = new URL("https://localhost:9595/pizzashack/1.0.0/menu");
+            HttpsURLConnection conn = (HttpsURLConnection) url.openConnection();
+            conn.setSSLSocketFactory(sslsocketfactory);
+            InputStream inputstream = conn.getInputStream();
+            System.out.println("Test is working properly ");
+            InputStreamReader inputstreamreader = new InputStreamReader(inputstream);
+            BufferedReader bufferedreader = new BufferedReader(inputstreamreader);
+            String string = null;
+            while ((string = bufferedreader.readLine()) != null) {
+                System.out.println("Received " + string);
+                break;
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            //e.printStackTrace();
+            String x = e.toString();
+            if (x.equalsIgnoreCase("javax.net.ssl.SSLHandshakeException: Received fatal alert: bad_certificate")) {
+                System.out.println("Test is working properly");
```

Comment:
Shall we please make this a log.info? Also please remove commented code segments.